src/dft.py

```diff
@@ -4,11 +4,28 @@
 from tree import T
 
 
+#TODO use other structure than lists?
+#TODO don't add leaves?
+#TODO switch not node to node
+
 def in_order(t: T | None) -> Iterable[int]:
     """In-order traversal of a tree.
 
     >>> tree = T(2, T(1, None, None), T(4, T(3, None, None), T(5, None, None)))
     >>> list(in_order(tree))
     [1, 2, 3, 4, 5]
     """
-    return  # FIXME
+    stack = [t]
+    output = []
+
+    while True:
+        node = stack.pop()
+        if not node:
```

Comment:
If `node` is the integer `0`, this `if not node` branch will be run. It's possible that you cannot have a value without a `None` above it in the stack, I can't quite see if that is the case, but *if* it is possible to pop zero here, then the code will break.